PeripheryFunctions/TS_Predict.m

```diff
@@ -0,0 +1,196 @@
+            useAllFeatures = true;
+        case 'topFeature'
+            useAllFeatures = false;
+    end
+else
+    % Otherwise, select the model with the best cross-validated accuracy
+    if exist('featureClassifier','var')
+      if exist('jointClassifier','var')
+        if any(featureClassifier.CVAccuracy > jointClassifier.CVAccuracy)
+          useAllFeatures = false;
+        end
+      else
+        useAllFeatures = false;
+      end
+    elseif ~exist('jointClassifier','var')
+      useAllFeatures = false;
+    end 
+end
+
+if useAllFeatures
+  myClassifier = jointClassifier;
+  myAcc = myClassifier.Accuracy;
+  fprintf('Using joint classifier (%i features) [acc=%.2f%% for %i classes].\n',...
+                length(jointClassifier.Operation.ID),...
+                myAcc, length(myClassifier.Mdl.ClassNames));
+else
+  myClassifier = featureClassifier;
+  myAcc = myClassifier.Accuracy;
+  
+  fprintf('Using classifier for feature "%s" (%i) [acc=%.2f%% for %i classes].\n',...
+                myClassifier.Operation.Name, myClassifier.Operation.ID,...
+                myAcc, length(myClassifier.Mdl.ClassNames));
+end
+
+myOps = myClassifier.Operation.ID;
+myMdl = myClassifier.Mdl;
+
+removeFile = false;
+
+toRecompute = true;
+if exist(predictionFilename,'file')
+    out = input(sprintf('Warning: %s already exists -- override? [yn] ',...
+                    predictionFilename),'s');
+    if out ~= 'y'
+        toRecompute = false;
+    end
+end
+    
+if toRecompute
+    if isempty(predictionFilename)
+        removeFile = true;
+        predictionFilename = 'tmp';
+        i = 1;
+        while exist([predictionFilename '.mat'],'file')
+            predictionFilename = sprintf('tmp-%i',i);
+            i = i + 1;
+        end
+        predictionFilename = [predictionFilename '.mat'];
+    end
+
+    keywords = cell(length(timeSeriesData),1);
+
+    [filePath,name] = fileparts(predictionFilename);
+
+    if ~isempty(filePath)
+        tsFilename = [filePath '/' name '_T.mat'];
+    else
+        tsFilename = [name '_T.mat'];
+    end
+
+    if exist(tsFilename,'file')
+        [~,name] = fileparts(tsFilename);
+        out = input(sprintf('Warning: %s already exists -- override? [yn] ',...
+                    name),'s');
+        if out ~= 'y'
+            return
+        end
+    end
+
+    save(tsFilename,'timeSeriesData','labels','keywords','-v7.3');
+    TS_Init(tsFilename,'','',0,predictionFilename);
+    delete(tsFilename);
+
+    TS_Compute(isParallel,[],myOps,'',predictionFilename,0);
+end
+
+if isfield(myClassifier,'normalizationInfo')
+    TS_Normalize(myClassifier.normalizationInfo.normFunction,...
+                    [0 1],...
+                    predictionFilename);
+                
+    if exist([predictionFilename(1:end-4) '_N.mat'],'file')
+        delete(predictionFilename);
+        predictionFilename = [predictionFilename(1:end-4) '_N.mat'];
+    end
+end
+
+whatData = load(predictionFilename);
```

Comment:
I didn't easily follow this because I don't have a use-case. Maybe put a usage example up top? Or can add a page on prediction to the _hctsa_ Gitbooks docs?

Reply:
(some commenting through would also be helpful)

Reply:
I think we'll need to put a use-case in the Gitbooks. There is a unit test in UnitTests/classifierUnitTest.m which is probably the best use case. We'll put that in Gitbooks and remove the unit test

Reply:
Awesome yep sounds good to me.